versions/development/SPEC.md

```diff
@@ -1049,6 +1057,43 @@ output {
 }
 ```
 
+#### Directory Outputs
+
+Directories can be declared as outputs just like files. For example:
+```
+output {
+  Directory d = "created/directory/"
+}
+```
+
+Notes:
+
+ - When a directory is output, the contents (including subdirectories) are considered part of the output but the path is not. 
```

Comment:
There seems to be a minor inconsistency between input and output.
```input: Directory = "/home/evan/bin" -> ".../workflow_wd/bin/..."
output: Directory = "bin" -> ".../workflow_output/?/..."```

I think I may be misreading. Is the intent that the name of the directory is lost (bin), or only the 'path' to the directory that contains the directory? It is confusing. I think the intent is the same behaviour as for Files, the name is kept, and the 'contents', but not anything else: path-to, permissions, whether it is a link or not etc. As it is not specified for Files, do we need to specify it for directories? or do we also need to specify it for Files?

Reply:
I think this is right - you want the directory name and everything inside it. Thanks for pointing this out, I'll try to reword things